src/generator/qbe.rs

```diff
@@ -0,0 +1,1221 @@
+                None => write!(f, "ret"),
+            },
+            Self::Jnz(val, if_nonzero, if_zero) => {
+                write!(f, "jnz {}, @{}, @{}", val, if_nonzero, if_zero)
+            }
+            Self::Jmp(label) => write!(f, "jmp @{}", label),
+            Self::Call(name, args) => {
+                write!(
+                    f,
+                    "call ${}({})",
+                    name,
+                    args.iter()
+                        .map(|(ty, temp)| format!("{} {}", ty, temp))
+                        .collect::<Vec<String>>()
+                        .join(", "),
+                )
+            }
+            Self::Alloc8(size) => write!(f, "alloc8 {}", size),
+            Self::Store(ty, dest, value) => {
+                if matches!(ty, QbeType::Aggregate(_)) {
+                    unimplemented!("Store to an aggregate type");
+                }
+
+                write!(f, "store{} {}, {}", ty, value, dest)
+            }
+            Self::Load(ty, src) => {
+                if matches!(ty, QbeType::Aggregate(_)) {
+                    unimplemented!("Load aggregate type");
+                }
+
+                write!(f, "load{} {}", ty, src)
+            }
+        }
+    }
+}
+
+/// QBE type
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[allow(dead_code)]
+enum QbeType {
+    // Base types
+    Word,
+    Long,
+    Single,
+    Double,
+
+    // Extended types
+    Byte,
+    Halfword,
+
+    /// Aggregate type with a specified name
+    Aggregate(String),
+}
+
+impl QbeType {
+    /// Returns a C ABI type. Extended types are converted to closest base
+    /// types
+    fn into_abi(self) -> Self {
+        match self {
+            Self::Byte | Self::Halfword => Self::Word,
+            other => other,
+        }
+    }
+
+    /// Returns the closest base type
+    fn into_base(self) -> Self {
+        match self {
+            Self::Byte | Self::Halfword => Self::Word,
+            Self::Aggregate(_) => Self::Long,
+            other => other,
+        }
+    }
+
+    /// Returns byte size for values of the type
+    fn size(&self) -> u64 {
+        match self {
+            Self::Word | Self::Single => 4,
+            Self::Long | Self::Double => 8,
+            Self::Byte => 1,
+            Self::Halfword => 2,
+
+            // Aggregate types are syntactic sugar for pointers ;)
+            Self::Aggregate(_) => 8,
+        }
+    }
+}
+
+impl fmt::Display for QbeType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Word => write!(f, "w"),
+            Self::Long => write!(f, "l"),
+            Self::Single => write!(f, "s"),
+            Self::Double => write!(f, "d"),
+
+            Self::Byte => write!(f, "b"),
+            Self::Halfword => write!(f, "h"),
+
+            Self::Aggregate(name) => write!(f, ":{}", name),
+        }
+    }
+}
+
+/// QBE value that is accepted by instructions
+#[derive(Debug, Clone)]
+#[allow(dead_code)]
```

Comment:
Can this (and all the other lint preventers) be removed?

Reply:
No, otherwise clippy will fail

Reply:
If it's not used, you can instead add an underscore (`Const` -> `_Const`).

Sorry for being so picky.😅

Reply:
I think this can be removed, but not the one in `QbeType` though since it will break `-D warnings` in Clippy. Note that this only applies to a specific enum and not the entire file, this is fine to have complete API

Reply:
Fair point